includes/data-port/class-sensei-data-port-utilities.php

```diff
@@ -53,12 +53,13 @@ public static function create_user( $username, $email = '', $role = '' ) {
 	 * Get an attachment by providing its source. The source can be a URL or a filename from the media library. If the
 	 * source is an external URL, it will be retrieved and an appropriate attachment will be created.
 	 *
-	 * @param string $source     Filename or URL.
-	 * @param int    $parent_id  Id of the parent post.
+	 * @param string $source             Filename or URL.
+	 * @param int    $parent_id          Id of the parent post.
+	 * @param array  $allowed_mime_types Allowed mime types.
 	 *
 	 * @return int|WP_Error  Attachment id on success, WP_Error on failure.
 	 */
-	public static function get_attachment_from_source( $source, $parent_id = 0 ) {
+	public static function get_attachment_from_source( $source, $parent_id = 0, $allowed_mime_types = null ) {
 		if ( false === filter_var( $source, FILTER_VALIDATE_URL ) ) {
```

Comment:
I think we need to check the mime type in this scenario. I believe it is stored in the attachment.

Reply:
Oops! My bad! I don't know how I miss that 😳
Fixed with a bunch of commits. Tomorrow I'll test a little more, but I think now it's working well.